### DISCUSSION TITLE
Errors When Deploying in Jamf - Intel & Apple Silicon

### DISCUSSION THREAD

**Discussion by jonidaher** [grahampugh/erase-install#200] [Q&A]
Hello, this is a great workflow, and I'm very grateful to Graham for providing it for us all. However, I'm having some issues on some devices. Hoping someone knows what's going on. 

I've been testing upgrading machines from Big Sur to Monterey using the **erase-install.sh** package with the following arguments:
**--force-curl --reinstall** 
The full line in Files & Processes is this: **/Library/Management/erase-install/erase-install.sh --force-curl --reinstall** 

**Intel machines** - Worked very well in three or four, but a couple keep returning this error: 

_**ERROR: downloaded installinstallmacos.py does not match checksum. Possible corrupted file. Deleting file.
Could not download installinstallmacos.py so cannot continue.
   [erase-install] 'caffeinate' ended**_

**Apple Silicon machines** - I've tested on two machines. The process asks for administrator credentials and then the screen-wide banner appears. However, the upgrade doesn't succeed. This is the error in Jamf: 

**_Preparing: 43.7%            
Preparing: 43.8%            
Preparing: 100.0%            An error occurred preparing the update.
   [erase-install] 'Self Service' ended
   [erase-install] 'caffeinate' ended
/Library/Management/erase-install/erase-install.sh: line 945: 37028 Terminated: 15          /usr/bin/caffeinate -dimsu -w $pid
   [erase-install] 'jamfHelper' ended
/Library/Management/erase-install/erase-install.sh: line 724: 37101 Terminated: 15          "$jamfHelper" -windowType $window_type -title "${!dialog_reinstall_title}" -heading "${!dialog_reinstall_heading}" -description "${!dialog_reinstall_desc}" -icon "$dialog_reinstall_icon"_**

I have run the policy several times on each device that has failed. Please let me know if you have any ideas. Thank you.

**Comment by grahampugh** [comment]
Make sure you have the latest version of the script to prevent the checksum error.

If you get the "error preparing the update", unfortunately you have to start again. It means the download got corrupted.

**Reply by jonidaher** [reply]
Please bear with me, as I'm relatively new to all this. I thought adding the --force-curl argument meant that I was always grabbing the latest script.

**Reply by grahampugh** [reply]
Do you have the latest erase-install.sh though?

**Reply by jonidaher** [reply]
I think so. I got it from here 7-10 days ago: https://github.com/grahampugh/erase-install/releases 
And as I mentioned, it does work on some machines.
